var/spack/repos/builtin/packages/vep-cache/package.py

```diff
@@ -0,0 +1,222 @@
+            cache_dir = join_path(species, f"{vep_version}_{assembly}")
+        else:
+            for check_species, assemblies in self.vep_species:
+                if species == check_species:
+                    assembly = assemblies[0]
+                    break
+            else:
+                assembly = ""
+            cache_dir = join_path(species, f"{vep_version}")
+
+        return {
+            "root": root,
+            "user_root": user_root,
+            "version": f"{cache_version}",
+            "type": f"{cache_type}",
+            "species": species,
+            "cache_species": species_cache,
+            "assembly": f"{assembly}",
+            "indexed": indexed,
+            "dir": cache_dir,
+            "full_path": join_path(root, cache_dir),
+        }
+
+    def url_for_version(self, version):
+        major = version.up_to(1)
+        url = f"https://raw.githubusercontent.com/Ensembl/ensembl-vep/release/{major}/INSTALL.pl"
+        vep = Spec(f"vep+installer@{major}")
+        return vep.package.vep_installer_path if vep.installed else url
+
+    def setup_run_environment(self, env):
+        if self.spec.satisfies("+env"):
+            cache = self.vep_cache_config(self.home)
+            env.set("VEP_OFFLINE", "1")
+            env.set("VEP_CACHE", "1")
+            env.set("VEP_DIR", cache["user_root"])
+            env.set("VEP_SPECIES", cache["species"])
+            env.set("VEP_CACHE_VERSION", cache["version"])
+            if cache["assembly"] is not None:
+                env.set("VEP_ASSEMBLY", cache["assembly"])
+            if cache["type"] == "refseq":
+                env.set("VEP_REFSEQ", "1")
+            if cache["type"] == "merged":
+                env.set("VEP_MERGED", "1")
+            if self.spec.satisfies("+fasta"):
+                pass
+
+    def cache_installer_args(self):
+        cache = self.vep_cache_config(self.prefix)
+        args = [
+            "--CACHEDIR",
+            cache["full_path"],
+            "--CACHE_VERSION",
+            cache["version"],
+            "--SPECIES",
+            cache["cache_species"],
+        ]
+        if cache["assembly"] is not None:
+            args += ["--ASSEMBLY", cache["assembly"]]
+
+        return args
+
+    def installer_args(self):
+        auto = "cf" if self.spec.satisfies("+fasta") else "c"
+        args = ["--AUTO", auto, "--NO_UPDATE", "--NO_TEST"]
+        args += self.cache_installer_args()
+        return args
+
+    def install_with_installer(self):
+        installer = which(self.vep.vep_installer_path)
+        installer(*self.installer_args())
+
+    def install(self, spec, prefix):
+        cache = self.vep_cache_config(self.prefix)
+        mkdirp(cache["full_path"])
+        if spec.satisfies("+installer"):
+            self.install_with_installer()
+        else:
+            tarball = _vep_cache_filename(
+                version=cache["version"], species=cache["species"], assembly=cache["assembly"]
```

Comment:
This needs the additional parameter source=cache["type"]:
```suggestion
                version=cache["version"], species=cache["species"], assembly=cache["assembly"], source=cache["type"]
```